django/contrib/auth/backends.py

```diff
@@ -119,6 +120,42 @@ def has_module_perms(self, user_obj, app_label):
             for perm in self.get_all_permissions(user_obj)
         )
 
+    def with_perm(self, perm, is_active=True, include_superusers=True, obj=None):
+        """
+        Return users that have permission "perm". By default, include active
+        users and all superusers.
+        """
+        if not isinstance(perm, (str, Permission)):
+            raise TypeError(
+                'The `perm` argument must be a string or a permission instance.'
+            )
+        elif isinstance(perm, str):
```

Comment:
Would we rather this were the following?
```python
if isinstance(perm, str):
    ...
elif not isinstance(perm, Permission):
    raise TypeError(...)
```

Reply:
Agreed.